code/modules/awaymissions/mission_code/ruins/telecomns.dm

```diff
@@ -0,0 +1,430 @@
+		qdel(CT)
+	kaboom.activate()
+	kaboom.icon_state = "death-bomb-active" // something funny goes on with icons here
+	qdel(flick_holder)
+	qdel(src)
+
+/obj/machinery/syndicatebomb/telecomns_doomsday_please_dont_spawn // They are going to spawn it on station anyway. I can feel it.
+	name = "\improper D.V.O.R.A.K's Doomsday Device"
+	icon_state = "death-bomb"
+	desc = "Nice to see AI's are improvising on the standard doomsday device. Good to have variety. Also probably a good idea to start running."
+	anchored = TRUE
+	timer_set = 100
+	payload = /obj/item/bombcore/telecomns_doomsday_please_dont_spawn
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+
+/obj/machinery/syndicatebomb/telecomns_doomsday_please_dont_spawn/singularity_act()
+	return // saves me headache later
+
+/obj/machinery/syndicatebomb/telecomns_doomsday_please_dont_spawn/ex_act(severity)
+	return // No.
+
+/obj/machinery/syndicatebomb/telecomns_doomsday_please_dont_spawn/screwdriver_act(mob/user, obj/item/I)
+	to_chat(user, "<span class='danger'>[src] is welded shut! You can't get at the wires!</span>")
+
+/obj/item/bombcore/telecomns_doomsday_please_dont_spawn
+	name = "a supermatter charged bomb core"
+	desc = "If you are looking at this, please don't put it in a bomb"
+
+/obj/item/bombcore/telecomns_doomsday_please_dont_spawn/ex_act(severity) // No
+	return
+
+/obj/item/bombcore/telecomns_doomsday_please_dont_spawn/burn() // Still no.
+	return
+
+/obj/item/bombcore/telecomns_doomsday_please_dont_spawn/detonate()
+	if(loc && istype(loc, /obj/machinery/syndicatebomb))
+		loc.invisibility = 90
+	for(var/turf/simulated/wall/indestructible/riveted/R in range(25, get_turf(src)))
+		R.ChangeTurf(/turf/space)
+	explosion(get_turf(src), 30, 40, 50, 60, 1, 1, 65, 0)
+	sleep(3 SECONDS)
+	var/obj/singularity/S = new/obj/singularity(get_turf(src))
+	S.consumedSupermatter = TRUE // woe large supermatter to eat the remains apon thee
+	S.energy = 4000
+	sleep(25 SECONDS)
+	S.visible_message("<span class='danger'>[S] collapses in on itself, vanishing as fast as it appeared!</span>")
+	qdel(S)
+	if(loc && istype(loc, /obj/machinery/syndicatebomb))
+		qdel(loc)
+	qdel(src)
+
+/turf/simulated/floor/catwalk/airless
+	oxygen = 0
+	nitrogen = 0
+	temperature = TCMB
+
+/obj/machinery/economy/vending/snack/trapped
+	aggressive = TRUE
+
+
+/mob/living/simple_animal/hostile/hivebot/strong/malfborg
+	name = "Security cyborg"
+	desc = "Oh god they still have access to these"
+	icon = 'icons/mob/robots.dmi'
+	icon_state = "Noble-SEC"
+	health = 200
+	maxHealth = 200
+	faction = list("malf_drone")
+	ranged = TRUE
+	rapid = 2
+	speed = 0.5
+	projectiletype = /obj/item/projectile/beam/disabler/weak
+	projectilesound = 'sound/weapons/taser2.ogg'
+	gold_core_spawnable = NO_SPAWN // Could you imagine xenobio with this? lmao.
+	a_intent = INTENT_HARM
+	var/obj/item/melee/baton/infinite_cell/baton = null // stunbaton bot uses to melee attack
```

Comment:
probably should implement some garbage collection signals for this :)

Reply:
ooh shit, thank you

Reply:
it already gets qdel'd on destroy